### ISSUE TITLE
SSL: WRONG_VERSION_NUMBER since upgrade to controller 5.5.14

### ISSUE THREAD

**Issue by SMCerina** [msweetser/check_unifi#5]
After upgrading to contoller 5.5.14 and cannot use the check anymore. 
I get the following error:
UNKNOWN: Error connecting to UniFi Controller (<urlopen error [SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:581)>)|Online=0 Offline=0
Exception urllib2.URLError: URLError(SSLError(1, u'[SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:581)'),) in <bound method Controller.__del__ of <unifi.controller.Controller instance at 0x766a51c0>> ignored

**Comment by luigixp** [comment]
I get the same error, just upgraded to 5.5.14. About to downgrade now.

**Comment by SMCerina** [comment]
I don’t think that downgrading is a solution. The unifi pip module is unmaintained and because of that the check_unifi will not working in the future.

**Comment by luigixp** [comment]
Haha yup, I'm finding that out now. So I'll just remove it. If anyone comes up with another solution please let me know.

**Comment by bigpresh** [comment]
My fix in PR #7 (changing check_unifi to use pyunifi) works for me and solves this issue.

**Comment by luigixp** [comment]
Able to write a quick guide on this please?

**Comment by luigixp** [comment]
It works, I do get the warning about https though. Is there a way to get rid of that message? I'm running 5.5.21. I rebooted an AP to see if I get the critical message when atleast one ap goes down.

:~ $ sudo ./check_unifi -S -H 10.0.0.1 -u user -p pass -v v5 -c 1
/usr/local/lib/python2.7/dist-packages/urllib3/connectionpool.py:858: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
CRITICAL: Offline: 1 (Closet AP) Online: 1 | Online=1 Offline=1

**Comment by bigpresh** [comment]
Yeah, I get that warning too, but Nagios appears to ignore it, so I didn't see it as a big problem - to be fair, disabling cert verification is something that shouldn't be done blithely, so a warning when you run the plugin directly about it isn't necessarily a bad thing :)

I'm not a Python coder, but It looks like disabling it would involve pulling in urllib3 and disabling it - https://stackoverflow.com/questions/27981545/suppress-insecurerequestwarning-unverified-https-request-is-being-made-in-pytho suggests:

```python
import urllib3
urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
```

I just wonder (a) if it's a good thing to disable it, and (b) if it should be done, because as it stands, the plugin code doesn't need to know or care how the pyunifi code talks to the Unifi controller, doing this requires that code to know about urllib3 etc, which is just a transport that it shouldn't need to know/care about.

**Comment by bigpresh** [comment]
No worries - wanted to get my APs monitored, so was fixing it for myself anyway, only fair to contribute it back :)

**Comment by demokom-gmail-com** [comment]
I decided this problem myself by adding two lines:
..
..
import ssl
..
..
..
ssl_version=ssl.PROTOCOL_SSLv3

And it works like a charm!

**Comment by nicnord** [comment]
How do you go about configuring the pyunifi to talk to the controller and check_unifi to talk to pyunifi? I am not very experienced in CentOS but have followed this guide and the pyunifi one just not quite sure how everything pieces together.

Any and all help you can provide would be much appreciated.

**Comment by thosch66** [comment]
> I decided this problem myself by adding two lines:

In which file and at which lines?
